libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts

```diff
@@ -37,9 +47,11 @@ const txToOps = (info: AccountShapeInfo, accountId: string, txs: CosmosTx[]): Co
   for (const tx of txs) {
     let fees = new BigNumber(0);
 
-    tx.tx.auth_info.fee.amount.forEach(elem => {
-      if (elem.denom === unitCode) fees = fees.plus(elem.amount);
-    });
+    if (tx.tx && tx.tx.auth_info && tx.tx.auth_info.fee && tx.tx.auth_info.fee.amount) {
+      tx.tx.auth_info.fee.amount.forEach(elem => {
+        if (elem.denom === unitCode) fees = fees.plus(elem.amount);
+      });
+    }
```

Comment:
This part
https://github.com/LedgerHQ/ledger-live/blob/3c88ce4b165b83d4f6614d535a2abc299940ffca/libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts#L48-L54

Could be replaced by optional chaining and the use of `reduce` for better code quality and readability.

```ts
    const fees: BigNumber = tx.tx?.auth_info?.fee?.amount?.reduce((acc, curr) => {
      if (curr.denom === unitCode) {
        return acc.plus(curr.amount);
      }
      return acc;
    }, new BigNumber(0));
```

Reply:
Not exactly what I had in mind. I can't use github suggestion feature because "Applying suggestions on deleted lines is not supported." but here is a patch with this suggestion (can be applied with `git apply /path/to/changes.patch`):

<details>
<summary>Diff</summary>

```
diff --git a/libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts b/libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts
index 5b5e93b1df..3f2238b1bd 100644
--- a/libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts
+++ b/libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts
@@ -45,13 +45,15 @@ const txToOps = (info: AccountShapeInfo, accountId: string, txs: CosmosTx[]): Co
   const unitCode = currency.units[1].code;
   const ops: CosmosOperation[] = [];
   for (const tx of txs) {
-    let fees = new BigNumber(0);
-
-    if (tx.tx && tx.tx.auth_info && tx.tx.auth_info.fee && tx.tx.auth_info.fee.amount) {
-      tx.tx.auth_info.fee.amount.forEach(elem => {
-        if (elem.denom === unitCode) fees = fees.plus(elem.amount);
-      });
-    }
+    const fees: BigNumber = tx.tx?.auth_info?.fee?.amount?.reduce(
+      (acc: BigNumber, curr: { denom: string; amount: string }) => {
+        if (curr.denom === unitCode) {
+          return acc.plus(curr.amount);
+        }
+        return acc;
+      },
+      new BigNumber(0),
+    );

     const op: CosmosOperation = getBlankOperation(tx, fees, accountId);
```

</details>



[changes.patch](https://github.com/LedgerHQ/ledger-live/files/12891152/changes.patch)


The jist of it is:
- remove the unnecessary use of `let`
- remove the long `if` condition and replace it with optional chaining